### PR TITLE
Feature/redbox 497

### PR DESCRIPTION
**Pull request by jamesrichards4**
## Context

A redbox user should use our chat tool unless they specifically request a different tool using keywords

## Changes proposed in this pull request

Removed all semantic routing and the dependency
Removed gratitude, coach and ability tools.
Replaced our name -> Runnable 'routes' list with a map of LangChain tools
Added a new endpoint to get all available tools

## Things to check

- [x] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [x] I have run integration tests

### CODE REVIEW

**Comment by andy-symonds** [comment]
@jamesrichards4 could do with a `make format lint` to fix ruff formating?

**Comment by jamesrichards4** [comment]
> @jamesrichards4 could do with a `make format lint` to fix ruff formating?

No changes with make format or make format lint on my end. What are you seeing that's out of style?

File: redbox-core/redbox/models/chain.py [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+    role: str
+    text: str
 
 
 class ChainInput(BaseModel):
     question: str = Field(description="The last user chat message")
-    file_uuids: list[UUID] = Field(description="List of files to process")
-    user_uuid: UUID = Field(description="User the chain in executing for")
-    chat_history: list[ChatMessage] = Field(description="All previous messages in chat (excluding question)")
+    file_uuids: list[str] = Field(description="List of files to process")
```
[the remark this answers is not in the document]
Comment (jamesrichards4) [reply redbox-core/redbox/models/chain.py]: we can, good call. There was a lot of faffing around with pydantic v1/v1 and langchain stuff here but reintroducing just this works

File: core-api/core_api/routes/chat.py [hunk 2]
```diff
@@ -32,10 +31,6 @@
     version="0.1.0",
     openapi_tags=[
         {"name": "chat", "description": "Chat interactions with LLM and RAG backend"},
-        {
-            "name": "embedding",
-            "description": "Embedding interactions with SentenceTransformer",
```
Comment (gecBurton) [inline core-api/core_api/routes/chat.py:37]: 👍

File: core-api/core_api/semantic_routes.py [hunk 3]
```diff
@@ -5,142 +5,60 @@
...
-        "What does Redbox do?",
-        "What can Redbox do",
-        "What don't you do",
-        "Please help me",
-        "Please help",
-        "Help me!",
-        "help",
-    ],
-)
+def as_chat_tool(
```
Comment (gecBurton) [inline core-api/core_api/semantic_routes.py:23]: nice and clear

**Review by andy-symonds — APPROVED** [review]
LGTM

**Review by gecBurton — APPROVED** [review]
less is more :)

**Pull request merged**